interfaces/builtin/network_status.go

```diff
@@ -24,131 +24,82 @@ import (
-   name="com.ubuntu.connectivity1.NetworkingStatus",
-
-# allow queries from unconfined
-dbus (receive)
-    bus=system
-    path=/com/ubuntu/connectivity1/NetworkingStatus{,/**}
-    interface=org.freedesktop.DBus.*
-    peer=(label=unconfined),
+# Description: allow providing network connectivity status
 `
 
 const networkStatusConnectedSlotAppArmor = `
-# Description: allow access to NetworkingStatus service
-
-dbus (receive)
-    bus=system
-    path=/com/ubuntu/connectivity1/NetworkingStatus{,/**}
-    interface=org.freedesktop.DBus.*
-    peer=(label=###PLUG_SECURITY_TAGS###),
+# Description: allow providing network connectivity status
 `
 
 const networkStatusConnectedPlugAppArmor = `
-# Description: Allow using NetworkingStatus service.
-
-#include <abstractions/dbus-strict>
+# Description: allow access to network connectivity status
 
-# Allow all access to NetworkingStatus service
-dbus (send)
-    bus=system
-    interface=com.ubuntu.connectivity1.NetworkingStatus{,*}
-    path=/com/ubuntu/connectivity1/NetworkingStatus
-    peer=(label=###SLOT_SECURITY_TAGS###),
+#include <abstractions/dbus-session-strict>
 
-dbus (send)
-    bus=system
-    path=/com/ubuntu/connectivity1/NetworkingStatus{,/**}
-    interface=org.freedesktop.DBus.*
+# Allow access to xdg-desktop-portal NetworkMonitor methods and signals
+dbus (send, receive)
+    bus=session
+    interface=org.freedesktop.portal.NetworkMonitor
+    path=/org/freedesktop/portal/desktop
     peer=(label=###SLOT_SECURITY_TAGS###),
 `
 
-const networkStatusPermanentSlotDBus = `
-<policy user="root">
-    <allow own="com.ubuntu.connectivity1.NetworkingStatus"/>
-    <allow send_destination="com.ubuntu.connectivity1.NetworkingStatus"/>
-</policy>
-
-<policy context="default">
-    <deny own="com.ubuntu.connectivity1.NetworkingStatus"/>
-    <allow send_destination="com.ubuntu.connectivity1.NetworkingStatus"/>
-</policy>
-
-<limit name="max_replies_per_connection">1024</limit>
-<limit name="max_match_rules_per_connection">2048</limit>
-`
-
-type networkStatusInterface struct{}
-
-func (iface *networkStatusInterface) Name() string {
-	return "network-status"
-}
-
-func (iface *networkStatusInterface) StaticInfo() interfaces.StaticInfo {
-	return interfaces.StaticInfo{
-		Summary:              networkStatusSummary,
-		BaseDeclarationSlots: networkStatusBaseDeclarationSlots,
-	}
+type networkStatusInterface struct {
+	commonInterface
 }
 
 func (iface *networkStatusInterface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
 	const old = "###SLOT_SECURITY_TAGS###"
-	new := slotAppLabelExpr(slot)
-	spec.AddSnippet(strings.Replace(networkStatusConnectedPlugAppArmor, old, new, -1))
+	var new string
+	if implicitSystemConnectedSlot(slot) {
+		new = "unconfined"
+	} else {
+		new = slotAppLabelExpr(slot)
+	}
+	snippet := strings.Replace(networkStatusConnectedPlugAppArmor, old, new, -1)
+	spec.AddSnippet(snippet)
```

Comment:
AppArmorConnectedPlug() can be removed entirely, adding `connectedPlugAppArmor: networkStatusConnectedPlugAppArmor` to commonInterface.